js/testapps/google-ai-code-execution/src/index.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { config } from 'dotenv';
+import * as z from 'zod';
+config();
+// Import the Genkit core libraries and plugins.
+import { generate } from '@genkit-ai/ai';
+import { configureGenkit } from '@genkit-ai/core';
+import { firebase } from '@genkit-ai/firebase';
+import { googleAI } from '@genkit-ai/googleai';
+
+// Import models from the Google AI plugin. The Google AI API provides access to
+// several generative models. Here, we import Gemini 1.5 Flash.
+import { gemini15Flash } from '@genkit-ai/googleai';
+
+// From the Firebase plugin, import the functions needed to deploy flows using
+// Cloud Functions.
+import { defineFlow } from '@genkit-ai/flow';
+
+configureGenkit({
+  plugins: [
+    // Load the Firebase plugin, which provides integrations with several
+    // Firebase services.
+    firebase(),
+    // Load the Google AI plugin. You can optionally specify your API key
+    // by passing in a config object; if you don't, the Google AI plugin uses
+    // the value from the GOOGLE_GENAI_API_KEY environment variable, which is
+    // the recommended practice.
+    googleAI(),
+  ],
+  // Log debug output to tbe console.
+  logLevel: 'debug',
+  // Perform OpenTelemetry instrumentation and enable trace collection.
+  enableTracingAndMetrics: true,
+});
+
+// Define a simple flow that prompts an LLM to generate menu suggestions.
+export const menuSuggestionFlow = defineFlow(
+  {
+    name: 'menuSuggestionFlow',
+    inputSchema: z.string(),
+    outputSchema: z.string(),
+  },
+  async (subject) => {
+    // Construct a request and send it to the model API.
+    const prompt = `Write and execute some code for generating the first 3 Fibonacci numbers.`;
+    const llmResponse = await generate({
+      model: gemini15Flash,
+      prompt: prompt,
+      config: {
+        temperature: 1,
+        codeExecution: true,
+      },
+    });
+
+    // Handle the response from the model API. In this sample, we just
+    // convert it to a string, but more complicated flows might coerce the
+    // response into structured output or chain the response into another
+    // LLM call, etc.
+    return llmResponse.text();
```

Comment:
can you please do a sample here where there's a back and forth? to demonstrate that history conversion happens correctly for custom parts?

Reply:
I've added the sample, it doesn't really feel nice using this feature to be honest, with all the `any`

Maybe a model should have some way of passing back a custom part schema to generate? I'm not sure how much work that would be, or how possible it would be

Reply:
I don't know, it's not terrible...  we can maybe add some helper functions to help extract code executions. As a follow up.